dynamiq/nodes/tools/jina.py

```diff
@@ -0,0 +1,242 @@
+        config = ensure_config(config)
+        self.run_on_node_execute_run(config.callbacks, **kwargs)
+
+        url = input_data.url or self.url
+        if not url:
+            logger.error(f"Tool {self.name} - {self.id}: failed to get input data.")
+            raise ValueError("URL is required for scraping")
+
+        headers = {
+            **self.connection.headers,
+            **({"X-Return-Format": self.response_format} if self.response_format != JinaResponseFormat.DEFAULT else {}),
+            "X-Timeout": str(self.timeout),
+        }
+
+        connection_url = self.SCRAPE_PATH + url
+
+        try:
+            response = self.client.request(
+                method=self.connection.method,
+                url=connection_url,
+                headers=headers,
+            )
+            response.raise_for_status()
+            scrape_result = response.text
+            if self.response_format in [JinaResponseFormat.PAGESHOT, JinaResponseFormat.SCREENSHOT]:
+                scrape_result = response.content
+        except Exception as e:
+            logger.error(f"Tool {self.name} - {self.id}: failed to get results. Error: {e}")
+
+            raise
+
+        if self.is_optimized_for_agents:
+            result = (
+                f"<Source URL>\n{input_data.url}\n<\\Source URL>"
+                f"\n<Scraped result>\n{scrape_result}\n<\\Scraped result>"
+            )
+        else:
+            result = {"url": input_data.url, "content": scrape_result}
+        logger.info(f"Tool {self.name} - {self.id}: finished with result:\n{str(result)[:200]}...")
+        return {"content": result}
+
+
+class JinaSearchInputSchema(BaseModel):
+    query: str = Field(..., description="Parameter to provide a search query.")
+    max_results: int = Field(
+        default=5,
+        ge=1,
+        le=100,
+        description="The maximum number of search results to return.",
+    )
+
+
+class JinaSearchTool(ConnectionNode):
+    """
+    A tool for performing web searches using the Jina AI API.
+
+    This tool accepts various search parameters and returns relevant search results.
+
+    Attributes:
+        group (Literal[NodeGroup.TOOLS]): The group to which this tool belongs.
+        name (str): The name of the tool.
+        description (str): A brief description of the tool.
+        connection (Jina): The connection instance for the Jina API.
+        include_images(bool): Whether include images in the search results.
+        input_schema (JinaSearchInputSchema): The input schema for the tool.
+        include_full_content(bool): Whether include full content of the search results.
+    """
+
+    group: Literal[NodeGroup.TOOLS] = NodeGroup.TOOLS
+    name: str = "Jina Search Tool"
+    description: str = "A tool for searching the web, powered by Jina."
+    SEARCH_PATH: ClassVar[str] = "https://s.jina.ai/"
+    connection: Jina
+    model_config = ConfigDict(arbitrary_types_allowed=True)
```

Comment:
Please move down `model_config = ConfigDict(arbitrary_types_allowed=True)` and divide with blank line